### PR TITLE
Upgrade @mdx/react to support react 18

### PR DESCRIPTION
**Pull request by rivajunior**
Issue:

## What I did

- Upgraded @mdx/react to v2.1.1 which supports react 18
- Updated the `test` script to accept ESM

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shilman** [comment]
thanks for your PR @rivajunior ! unfortunately, this is a breaking change so we'll need to tackle this as part of our 7.0 push. i'll look into what it will take to merge once 6.5 is released.

**Comment by yannbf** [comment]
Sensacional ver contribuições de um conterrâneo @rivajunior 🙌

**Comment by rivajunior** [comment]
> Sensacional ver contribuições de um conterrâneo @rivajunior 🙌

@yannbf quero cada vez mais contribuir com open source. Acho que só tenho que fazer virar hábito.

File: package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "run-chromatics": "node -r esm ./scripts/run-chromatics.js",
     "serve-storybooks": "http-server ./built-storybooks -p 8001",
     "start": "yarn workspace official-storybook storybook --no-manager-cache",
-    "test": "NODE_OPTIONS=--max_old_space_size=4096 jest",
+    "test": "NODE_OPTIONS=\"--max_old_space_size=4096 --experimental-vm-modules\" jest",
```
Comment (ndelangen) [inline package.json:77]: wait WHUT?

Is this needed?

Comment (rivajunior) [reply package.json:78]: Now @mdx/react is ESM only. See [Migrating from v1 to v2](https://mdxjs.com/migrating/v2/#esm).

Using `--experimental-vm-modules` is one of the options given by jest [docs](https://jestjs.io/docs/ecmascript-modules).

Comment (ndelangen) [reply package.json:78]: Ow I see... 👍

**Comment by ndelangen** [comment]
@rivajunior could you help getting the PR green? I merged in the latest changes on `next`...

**Comment by ndelangen** [comment]
Let me know if you need help!

**Comment by IanVS** [comment]
Thanks for your work on this, @rivajunior

I think this can be closed now that storybook 7 uses mdx2 by default, right @shilman?

**Comment by shilman** [comment]
Good call @IanVS. We're focused on getting 7.0 out with full MDX2 support. We can do patches back to 6.5 for specific bugfixes, but maintaining two versions of MDX is a tricky proposition anyway, so it wasn't a huge surprise that 6.5's "experimental" MDX2 support has been problematic.

**Pull request closed without merging**